src/reducers/monitors.js

```diff
@@ -0,0 +1,27 @@
+const UPDATE_MONITORS = 'scratch-gui/monitors/UPDATE_MONITORS';
+
+const initialState = {
+    monitors: []
+};
+
+const reducer = function (state, action) {
+    if (typeof state === 'undefined') state = initialState;
+    switch (action.type) {
+    case UPDATE_MONITORS:
+        return Object.assign({}, state, {monitors: action.monitors});
```

Comment:
Whoops misread this one.  The way you're doing it is how I was expecting. Except that there's the extra level of `monitors`. Since we don't have any other state to store, I think we can flatten this to a simple list.